src/utils/data/application-data.js

```diff
@@ -0,0 +1,22 @@
+const convertData = (data) => {
+  if (data.organizationId) {
+    return {
+      id: data.applicationId,
+      orgId: data.organizationId._id,
+      ...data,
+    };
+  }
+  return {
+    id: data.applicationId,
+    ...data,
+  };
+};
+
+export default (data) => {
+  if (data.length && data[0].applicationId && data.constructor === Array) {
```

Comment:
Although this works it might be a bit too verbose and unnecessary. Also, it is NOT to be trusted.

When data.length is true, it can mean one of two things:

1. The length property of the data object is truthy. (You can see how this can be untrustworthy).
2. The data object is an array (But because of the first issue above, it can't be trusted).

Example:
```
const object = {
    length: "I'm just a string with my property name as length".
}
if (object.length) {
    // This will be executed.
    console.log("Do you think its really an array now");
}
```

So I would strongly recommend that you don't check if its an array by checking if the length property is truthy. You run the risk of objects that seem to be an array but aren't.

The second thing I would like to bring your attention to is `data.constructor === Array`. This can also be spoofed because the `data.constructor` property can be updated. So this property is also untrustworthy.

Example:
```
const array = [];
array.constructor = null;

console.log(array.constructor === Array); // false
```

My suggestion: use Array.isArray()
```
const object = { length: 5 };
object.constructor = Array;

const array = [];
array.constructor = null;

console.log(Array.isArray(object)); // false
console.log(Array.isArray(array)); // true
```